resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -483,6 +483,7 @@ public DnsNameResolver(
         b.group(executor());
         b.channelFactory(channelFactory);
         b.option(ChannelOption.DATAGRAM_CHANNEL_ACTIVE_ON_REGISTRATION, true);
+        b.option(ChannelOption.SO_REUSEADDR, true);
```

Comment:
I think we should only do this if the localAddress is not null and the port is non-zero.

Reply:
done